src/auth/user-record-internal.ts

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { 
+  MultiFactorInfo, PhoneMultiFactorInfo
```

Comment:
How about moving these types into a new module (say `multi-factor`) and break the cyclic dependency?

Reply:
Thanks, good point, though had to pull in a few more members to a `multi-factor-internal` to get it to be acyclic.